### ISSUE TITLE
Extra text when rendering a line of a paragraph

### ISSUE THREAD

**Issue by gmile** [ManimCommunity/manim#2366]
## Description of bug / unexpected behavior

When rendering first line of a paragraph, extra text is rendered. See "Additional media files" section.

## Expected behavior

When rendering first line of a paragraph, only text from first line is rendered.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
from manim import *

class MyMovie(Scene):
    def construct(self):
        paragraph = Paragraph("First line", "Second line", "Third line")
        self.play(Create(paragraph[0]))
        self.interactive_embed()
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

![image](https://user-images.githubusercontent.com/207112/145084879-630633a2-764c-4a2e-9f5a-32d4174ddbb1.png)

</details>

## Additional comments

It works perfectly fine with `Code` object:

```python
from manim import *

class MyCode(Scene):
    def construct(self):
        code = '''line1 = 1
line2 = 2
line3 = 3
'''
        rendered_code = Code(code=code, tab_width=4, background="window",
                            language="Text", font="Monospace")
        self.play(Create(rendered_code[2][0]))
        self.interactive_embed()
```

**Comment by Viicos** [comment]
Not to harm the original writer(s) of this class, but imo it is poorly structured and needs a complete rewrite (the `gen_chars` method is never used, `__len__` is used instead of `len`, `for _ in range(len())` used, chars and alignment attributes are stored in the same `lines` attribute, etc). I'll try to do a rewrite and do a PR asap to get feedbacks.

**Comment by behackl** [comment]
> Not to harm the original writer(s) of this class, but imo it is poorly structured and needs a complete rewrite (the `gen_chars` method is never used, `__len__` is used instead of `len`, `for _ in range(len())` used, chars and alignment attributes are stored in the same `lines` attribute, etc). I'll try to do a rewrite and do a PR asap to get feedbacks.

No worries; the text classes have not been touched in quite a while and pretty much everyone agrees that they need to be rewritten. One thing to be aware of though: @naveen521kk has proposed a replacement of `Text` in #2355 that might make `Paragraph` obsolete.

**Comment by Viicos** [comment]
Oh you're right forgot about this recently opened issue. The op said that he wants to create an alias for `Paragraph` (8. `Paragraph` will be aliased to `Text`) so I'll wait for the moment.

**Comment by chopan050** [comment]
It seems like this issue was fixed by PR #2721. I'll close this issue.
